middleware.js

```diff
@@ -1,4 +1,38 @@
 var http = require('http');
 
 exports.createServer = function () {
-};
+  var middlewares = [];
+
+  function next(req, res, index) {
+    if(index == undefined) {
+      index = -1;
+    }
+    return function () {
+      for(var i = index + 1; i < middlewares.length; i++) {
+        if(middlewares[i].path == '*' || middlewares[i].path == req.url) {
+          middlewares[i].action.call(this, req, res, next(req, res, i));
+          return;
+        }
+      }
+    }
+  }
+
+  var server = http.createServer(function (req, res) {
+    next(req, res)();
+  });
+
+  http.Server.prototype.use = function () {
```

Comment:
Inheriting from `http.Server` here is not recommended in practice as it's confusing (because we are not writing `http.Server`) and potentially problematic (as for now the `server` contains every ~~redundant~~ methods from `http.Server`, but what we need here are just three methods: `use()`, `createServer()` and `listen()`.